File: leetcode-algorithms/008. String to Integer (atoi)/solution.py
def myAtoi(str):
	ans = 0
	maxint = 0x7fffffff
	minint = -2147483648
	maxVal = 0x100000000

	flag = False

	st = 0

	while st<len(str) and str[st]==' ':
		st += 1
	if st<len(str) and str[st]=='+':
		st += 1
	else:
		if st<len(str) and str[st]=='-':
			flag = True
			st += 1
	for i in range(st,len(str)):
		if str[i]<='9' and str[i]>='0':
			ans = ans * 10 + int(str[i])
			if ans>maxVal:
				ans = maxVal
		else:
			break

	if flag:
		ans = -ans

	if ans > maxint:
		ans = maxint
	if ans < minint:
		ans = minint

	return ans

if __name__ == '__main__':
	s = '234'
	res = myAtoi(s)

	print(res)

File: leetcode-algorithms/008. String to Integer (atoi)/solution.cpp
#include <iostream>
#include <string>
using namespace std;


int myAtoi(string str)
{
	long long ans = 0;
	const int maxint = 0x7fffffff;
	const int minint = 0x80000000;
	const long long max = 0x100000000;

	bool flag = false;

	int st = 0;

	while(st<str.length() && str[st]==' ')
		st++;
	if(st<str.length() && str[st]=='+')
		st++;
	else
	{
		if(st<str.length() && str[st]='-')
			flag = true,st++;
	}

	for(int i=st;i<str.length();i++)
	{
		if(str[i]<='9' && str[i]>='0')
		{
			ans = ans*10+str[i]-'0';
			if(ans>max)
				ans = max;
		}
		else
			break;
	}

	if(flag)
		ans = -ans;
	if(ans<minint)
		ans = minint;
	if(ans>maxint)
		ans = maxint;

	return ans;

}
int main()
{
	return 0;
}